Error when building with MinGW : ` ./copy_stubs.c:369:7: error: request for member 'ByteCount' in something not a structure or union`

I'm the maintainer of the Opam package for unison, and I am trying to publish the package for Unison 2.53.8.

Things seems working, except for Windows, with the MinGW port. I get the following errors: 

```
   # ./copy_stubs.c:369:7: error: request for member 'ByteCount' in something not a structure or union
  #   369 |   dupl.ByteCount.QuadPart = Long_val(len) +
  #       |       ^
  # ./copy_stubs.c:384:31: error: request for member 'TargetFileOffset' in something not a structure or union
  #   384 |   newTargetLen.QuadPart = dupl.TargetFileOffset.QuadPart + Long_val(len);
  #       |                               ^
  # make[2]: *** [Makefile.OCaml:208: copy_stubs.o] Error 2
  # make[2]: *** Waiting for unfinished jobs....
  # make[2]: Leaving directory '/d/opamroot/default/.opam-switch/build/unison.2.53.8/src'
  # make[1]: *** [Makefile:42: all] Error 2
  # make[1]: Leaving directory '/d/opamroot/default/.opam-switch/build/unison.2.53.8/src'
  # make: *** [Makefile:22: src] Error 2
```

(See https://github.com/ocaml/opam-repository/actions/runs/19137603433/job/54693594142#step:14:88)

I have no Windows machine to test, but this seems related to https://github.com/mingw-w64/mingw-w64/issues/90, which added this type definition to MinGW. But looking at `copy_stubs.c`, it seems like it should be handled by Unison if MinGW is too old...

Any idea what's happening? Of cours, I could just disable the support for Windows, but this is lame.

--- a/src/copy_stubs.c
+++ b/src/copy_stubs.c
@@ -15,6 +15,25 @@
     along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
+#ifdef _WIN32
+
+/* Minimum Windows API needed for getting the definition of
+ * struct DUPLICATE_EXTENTS_DATA with MinGW headers under MSYS2.
+ * As of 2025-11, MSYS2 by default targets Windows 7, which is
+ * not sufficient in this case. */
+#define WINVER 0x0603
+#define _WIN32_WINNT 0x0603  /* Windows 8.1 */
+
+#ifndef UNICODE
+#define UNICODE
+#endif
+#ifndef _UNICODE
+#define _UNICODE
+#endif
+
+#endif /* _WIN32 */
+
+
 #include <caml/memory.h>
 #include <caml/threads.h>
 #include <caml/unixsupport.h>
@@ -243,13 +262,6 @@ CAMLprim value unison_copy_file(value in_fd, value out_fd, value in_offs, value 
 
 #elif defined(_WIN32)
 
-
-#ifndef UNICODE
-#define UNICODE
-#endif
-#ifndef _UNICODE
-#define _UNICODE
-#endif
 
 #include <windows.h>
 #include <winioctl.h>

